distributed/scheduler.py

```diff
@@ -1464,10 +1487,9 @@ def state(self) -> TaskStateState:
 
     @state.setter
     def state(self, value: TaskStateState) -> None:
-        self.group.states[self._state] -= 1
-        self.group.states[value] += 1
+        self.group.transition(self._state, value)
+        self.prefix.transition(self._state, value)
         self._state = value
-        self.prefix.state_counts[value] += 1
 
```

Comment:
Something obvious: I think this is the only truly performance critical section since this is effectively called on every task state transition. The duration and nbytes update is only called whenever a task completes so at least five times more rarely. Since this essentially performs the same operations (plus a couple of additional function indirections/calls) this _should_ perform similarly to the old version

Reply:
similarly, the nbytes and duration updates should now be twice as expensive but that's a constant. Given that we iterated over everything and performed this computation every 100ms before I assume this is a net positive